From ba143a3c91cbc4f6081546c58ff828defc3daf2b Mon Sep 17 00:00:00 2001
From: Jean-Michel Bachot <jean-michel.bachot@intel.com>
Date: Fri, 12 Jul 2013 15:04:36 +0200
Subject: [PORT FROM MAIN] wifi: add custom android property for process pid

BZ: 99166 111626

A custom propoerty is created to store wpa_supplicant PID.
This property is used by the wifi HAL  to send a
SIGTERM signal to wpa_supplicant when stopping down wpa_supplicant.
This allows a nice shutdown of wpa_supplicant with clean interfaces
removal.

Change-Id: Ifd8d86f8c8f0524df8d193ea6ade73a89067f90d
Orig-Change-Id: I23c2019fcfb2dd2b344d86e0f860aec0736c9e69
Category: device enablement
Domain: CWS.WIFI-Common
Origin: upstream
Upstream-Candidate: no,proprietary
Signed-off-by: Jean-Michel Bachot <jean-michel.bachot@intel.com>
---
 wpa_supplicant/main.c |   18 ++++++++++++++++++
 1 files changed, 18 insertions(+), 0 deletions(-)

diff --git a/wpa_supplicant/main.c b/wpa_supplicant/main.c
index 39b837e..d0b9d97 100644
--- a/wpa_supplicant/main.c
+++ b/wpa_supplicant/main.c
@@ -16,6 +16,10 @@
 #include "driver_i.h"
 #include "p2p_supplicant.h"
 
+#ifdef ANDROID
+#include <cutils/properties.h>
+#endif
+
 extern struct wpa_driver_ops *wpa_drivers[];
 
 
@@ -316,6 +320,20 @@ int main(int argc, char *argv[])
 #endif /* CONFIG_P2P */
 	}
 
+#ifdef ANDROID
+	/* Write PID to system property (used by wifi HAL) */
+	char pidpropval[PROPERTY_VALUE_MAX];
+	int ret;
+	ret = os_snprintf(pidpropval, sizeof(pidpropval), "%d", getpid());
+	if (ret < 0 || (unsigned int) ret >= PROPERTY_VALUE_MAX) {
+		wpa_printf(MSG_ERROR, "property value invalid: %d", getpid());
+	}
+	if (property_set("wpa_supplicant.pid", pidpropval) < 0)
+		wpa_printf(MSG_ERROR, "property set failed: %s", pidpropval);
+	else
+		wpa_printf(MSG_DEBUG, "property set success: %s", pidpropval);
+#endif
+
 	if (exitcode == 0)
 		exitcode = wpa_supplicant_run(global);
 
-- 
1.7.4.1

